File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/user/DefaultKinveyClientCallback.java
package com.kinvey.androidTest.store.user;

import com.kinvey.android.model.User;
import com.kinvey.java.core.KinveyClientCallback;

import java.util.concurrent.CountDownLatch;

public class DefaultKinveyClientCallback implements KinveyClientCallback<User> {

    private CountDownLatch latch;
    User result;
    Throwable error;

    DefaultKinveyClientCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(User user) {
        this.result = user;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    private void finish() {
        latch.countDown();
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/user/DefaultKinveyClientCallback.kt
package com.kinvey.androidTest.store.user

import com.kinvey.android.model.User
import com.kinvey.java.core.KinveyClientCallback
import java.util.concurrent.CountDownLatch

class DefaultKinveyClientCallback(private val latch: CountDownLatch) : KinveyClientCallback<User> {

    var result: User? = null
    var error: Throwable? = null

    override fun onSuccess(result: User?) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable?) {
        this.error = error
        finish()
    }

    private fun finish() {
        latch.countDown()
    }
}